### ISSUE TITLE
Invalid "mismatched type" error for enum with #[repr]

### ISSUE THREAD

**Issue by netvl** [intellij-rust/intellij-rust#2477]
Suppose I have an enum like this:
```rust
// generated by bindgen, for example
pub type ComeCLib_Error = i32;
pub const SOMECLIB_ERROR_WHATEVER: SomeCLib_Error = -1234;

#[repr(i32)]
enum Error {
    Whatever = SOMECLIB_ERROR_WHATEVER,
}
```
Here IntelliJ-Rust shows a "mismatched type" error on the `Whatever` enum variant definition, "expected `isize`, found `i32`". However, it is not correct, because the `#[repr(i32)]` attribute overrides the underlying type, and rustc complains if e.g. an `as isize` cast is added.

Ideally, IntelliJ-Rust should detect the expected type for enum variant constants from the `#[repr(..)]` attribute and require values of these types when defining these constants.

**Comment by Undin** [comment]
Fixed in #2508
